storage/database/dynamodb.go

```diff
@@ -0,0 +1,477 @@
+}
+
+func (dynamo *dynamoDB) Close() {
+	close(dynamo.quitCh)
+}
+
+func (dynamo *dynamoDB) Meter(prefix string) {
+	// TODO-Klaytn: add metrics
+	dynamo.batchWriteTimeMeter = metrics.NewRegisteredMeter(prefix+"batchwrite/time", nil)
+	dynamo.batchWriteCountMeter = metrics.NewRegisteredMeter(prefix+"batchwrite/count", nil)
+	dynamo.batchWriteSizeMeter = metrics.NewRegisteredMeter(prefix+"batchwrite/size", nil)
+	dynamo.batchWriteSecPerItemMeter = metrics.NewRegisteredMeter(prefix+"batchwrite/secperitem", nil)
+	dynamo.batchWriteSecPerByteMeter = metrics.NewRegisteredMeter(prefix+"batchwrite/secperbyte", nil)
+}
+
+func (dynamo *dynamoDB) NewIterator() Iterator {
+	return nil
+}
+
+func (dynamo *dynamoDB) NewIteratorWithStart(start []byte) Iterator {
+	return nil
+}
+
+func (dynamo *dynamoDB) NewIteratorWithPrefix(prefix []byte) Iterator {
+	return nil
+}
+
+func (dynamo *dynamoDB) createBatchWriteWorker() {
+	failCount := 0
+	tableName := dynamo.config.TableName
+	batchWriteInput := &dynamodb.BatchWriteItemInput{
+		RequestItems: map[string][]*dynamodb.WriteRequest{},
+	}
+	dynamo.logger.Info("generate a dynamoDB batchWrite worker", "time", time.Now())
+
+	for {
+		select {
+		case <-dynamo.quitCh:
+			dynamo.logger.Info("close a dynamoDB batchWrite worker", "time", time.Now())
+			return
+		case batchInput := <-dynamo.writeCh:
+			batchWriteInput.RequestItems[tableName] = batchInput.items
+
+			BatchWriteItemOutput, err := dynamo.db.BatchWriteItem(batchWriteInput)
+			numUnprocessed := len(BatchWriteItemOutput.UnprocessedItems)
+			for err != nil || numUnprocessed != 0 {
+				if err != nil {
+					failCount++
+					dynamo.logger.Warn("dynamoDB failed to write batch items", "err", err, "failCnt", failCount)
+					if failCount > dynamoMaxRetry {
+						dynamo.logger.Error("dynamoDB failed many times. sleep a second and retry",
+							"failCnt", failCount)
+						time.Sleep(time.Second)
+					}
+				}
+
+				if numUnprocessed != 0 {
+					dynamo.logger.Debug("dynamoDB batchWrite remains unprocessedItem",
+						"numUnprocessedItem", numUnprocessed)
+					batchWriteInput.RequestItems[tableName] = BatchWriteItemOutput.UnprocessedItems[tableName]
+				}
+
+				BatchWriteItemOutput, err = dynamo.db.BatchWriteItem(batchWriteInput)
+				numUnprocessed = len(BatchWriteItemOutput.UnprocessedItems)
+			}
+
+			failCount = 0
+			batchInput.wg.Done()
+		}
+	}
+}
+
+func (dynamo *dynamoDB) NewBatch() Batch {
+	return &dynamoBatch{db: dynamo, tableName: dynamo.config.TableName, wg: &sync.WaitGroup{}}
+}
+
+type dynamoBatch struct {
+	db         *dynamoDB
+	tableName  string
+	batchItems []*dynamodb.WriteRequest
+	size       int
+	wg         *sync.WaitGroup
+}
+
+func (batch *dynamoBatch) Put(key, val []byte) error {
+	data := DynamoData{Key: key, Val: val}
+	dataSize := len(val)
+	if dataSize == 0 {
+		return nil
+	}
+
+	// If the size of the item is larger than the limit, it should be handled in different way
+	if len(key)+dataSize > dynamoWriteSizeLimit {
```

Comment:
```suggestion
	if len(marshaledData) > dynamoWriteSizeLimit {
```
I think the checking is better placed after the marshaling.